third_party/terraform/website/docs/d/datasource_google_secretmanager_secret_version.html.markdown

```diff
@@ -0,0 +1,49 @@
+---
+subcategory: "Cloud Platform"
+layout: "google"
+page_title: "Google: google_secretmanager_secret_version"
+sidebar_current: "docs-google-datasource-secretmanager-secret-version"
+description: |-
+  Get a Secret Manager secret's version.
+---
+
+# google\_secretmanager\_secret\_version
+
+Get a Secret Manager secret's version. For more information see the [official documentation](https://cloud.google.com/secret-manager/docs/) and [API](https://cloud.google.com/secret-manager/docs/reference/rest/v1beta1/projects.secrets.versions).
+
+## Example Usage
+
+```hcl
+data "google_secretmanager_secret_version" "basic" {
+  provider = google-beta
+  secret = "my-secret"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `project` - (Optional) The project to get the secret version for. If it
+    is not provided, the provider project is used.
+
+* `secret` - (Required) The secret to get the secret version for.
+
+* `version` - (Optional) The version of the secret to get. If it
+    is not provided, the latest version is retrieved.
+
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `secret_data` - The secret data. No larger than 64KiB. A base64-encoded string.
```

Comment:
Why do we have a data source that exposes `secret_data` when we are using `ignore_read` on the field in the TF resource? This field corresponds to `SecretVersion.payload.data` right?

Reply:
Ah, read the ds more closely. We make a separate call to `:access` to retrieve the data?

Reply:
Yea.  I wasn't exactly sure how to handle this in the resource, and it's the same case with `:enable` and `:disable`, they're all separate calls.  I'm going to spend some more time looking into it, but if you have any suggestions, please let me know!